java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -113,7 +113,15 @@ private static ArrowType getArrowType(Type type) {
     schema.getObjectProps().forEach((k,v) -> metadata.put(k, v.toString()));
 
     if (type == Type.RECORD) {
-      throw new UnsupportedOperationException();
+      for (Schema.Field field : schema.getFields()) {
+        final ArrowType arrowType = getArrowType(field.schema().getType());
+        if (arrowType != null) {
+          final FieldType fieldType = new FieldType(true, arrowType, null, null);
+          List<Field> children = null;
+          //TODO support complex type
+          arrowFields.add(new Field(field.name(), fieldType, children));
+        }
```

Comment:
should we throw on the else?

Reply:
Thanks, getArrowType won't return null, so I just removed if condition here.